### ISSUE TITLE
Dice decreases during training with Primus

### ISSUE THREAD

**Issue by slee-prenuvo** [MIC-DKFZ/nnUNet#3029]
Hello, I’m using the nnssl segmentation adaptation framework in the `nnssl_finetuning_inclusion` branch for 3D T1 brain MRI segmentation (~30 ROIs, PrimusM, batch size 2, patch size 160x160x160). The trainer I'm using is [PretrainedTrainer_Primus](https://github.com/TaWald/nnUNet/blob/nnssl_finetuning_inclusion/nnunetv2/training/nnUNetTrainer/pretraining/pretrainedTrainer.py), and I'm finetuning the pretrained model [PrimusM-OpenMind-MAE](https://huggingface.co/AnonRes/PrimusM-OpenMind-MAE).

I am seeing a consistent failure mode across several learning rates, where a few classes learn well, but several others either never improve or improve early and then collapse later in training.

#### 1) training_log1.txt and progress1.png:

In this run, I changed `self.initial_lr` from the deafult `1e-4` to `3e-5`. With both `1e-4` and `3e-5`, I observed similar behavior:
- One ROI Dice rises early (for example, up to 0.8887 at epoch 21) but later collapses badly (down to 0.0575 by epoch 207).
- A second ROI Dice rises and stays high throughout training.
- The last two ROI Dice scores remain close to zero for almost the entire run.
- The EMA pseudo-Dice peaks around 0.41, then gradually drops to about 0.36 and stays there.

#### 2) training_log2.txt and progress2.png:
In this run, I reduced `self.initial_lr` further from `3e-5` to `3e-6`. I noticed some improvements in that the last two ROI Dice scores increased to about 0.17 and 0.36, respectively, and the EMA pseudo-Dice increased to around 0.57. However, same as the above experiment, several ROIs stopped improving after some point and their Dice scores decreased again later in training.

#### 3) training_log3.txt and progress3.png:
This is a very low LR setup where I lowered `self.initial_lr` to `1e-6`. I also reduced warm-up phase from 50 to 10 epochs. This setup looked healthier early on:
- The EMA pseudo-Dice improved almost every epoch at first.
- It reached about 0.62 by epoch 180.

But eventually I still saw the same pattern:
- The EMA pseudo-Dice later dropped back toward 0.4.
- Some strong classes stayed in the high 0.8–0.94 range.
- A subset of other classes gradually drifted down toward zero.

To give you additional context
- Standard nnUNet trains successfully on the same dataset; EMA pseudo-Dice goes above 0.95.
- The ROIs that collapse are not necessarily very small structures. For example, the caudate is among them.
- I turned off mirroring, following the recommendation for left/right ROI tasks. I also tried turning mirroring back on, but it did not make much difference.
- I tested the same pretrained model and trainer on a different task: an easier binary whole-brain segmentation problem. In that setting, training behaved normally and the EMA pseudo-Dice improved up to about 0.95.

Has anyone seen this before? I'm unsure what the most sensible next steps would be. If anyone could advise or share similar experiences, I would really appreciate it. Thank you!

[training_log1.txt](https://github.com/user-attachments/files/27176316/training_log1.txt)
<img width="500" height="900" alt="Image" src="https://github.com/user-attachments/assets/ae0a6511-2a6b-400c-8496-ce2ce54bc15e" />

[training_log2.txt](https://github.com/user-attachments/files/27176572/training_log2.txt)
<img width="500" height="900" alt="Image" src="https://github.com/user-attachments/assets/53915040-f4cd-4353-b0b6-4b12bd008b91" />

[training_log3.txt](https://github.com/user-attachments/files/27176317/training_log3.txt)
<img width="500" height="900" alt="Image" src="https://github.com/user-attachments/assets/e514f120-8728-4e1c-8a99-84055f6fdd61" />

**Comment by FabianIsensee** [comment]
Tagging @TaWald and @constantinulrich as the experts on this matter 🫶

**Comment by TaWald** [comment]
Q1: Is this a consistent failure case across multiple datasets or just this one?
Q2: What `torch` version are you using?

Because I used to have the same error a while ago with an environment that had a broken`torch` version.
That was `torch 2.9`, I believe.

**Comment by FabianIsensee** [comment]
Good point, that was not 2.9 (2.9 was just slow) but 2.10+. If you are still on an old version of nnU-Net torch.compile enabled models will not converge properly with torch>=2.10. Try `nnUNet_compile=f` or make sure you are on the latest nnU-Net (or master)

**Comment by TaWald** [comment]
@slee-prenuvo What's the status?
Did you change your torch version and did this still persist?

**Comment by TaWald** [comment]
Closing this due to inactivity. Feel free to reopen if still relevant.
